src/sql/workbench/contrib/query/browser/queryEditor.ts

```diff
@@ -260,9 +260,10 @@ export class QueryEditor extends BaseEditor {
 		let content: ITaskbarContent[];
 		const previewFeaturesEnabled = this.configurationService.getValue('workbench')['enablePreviewFeatures'];
 		let connectionProfile = this.connectionManagementService.getConnectionProfile(this.input?.uri);
+		let fileExtension = this.input?.uri.substr(this.input?.uri.lastIndexOf('.') + 1);
 
 		// TODO: Make it more generic, some way for extensions to register the commands it supports
-		if (connectionProfile?.providerName === 'KUSTO') {
+		if (connectionProfile?.providerName === 'KUSTO' || fileExtension === 'kql' || fileExtension === 'kusto') {
```

Comment:
You should use https://github.com/microsoft/azuredatastudio/blob/main/src/vs/editor/common/services/modeService.ts#L40 to get the supported extensions so you don't have to update this if you add more later on.

Reply:
Makes sense. Made this change